flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/async/CompletedOperationCache.java

```diff
@@ -193,24 +209,34 @@ void cleanUp() {
     /** Stores the result of an asynchronous operation, and tracks accesses to it. */
     private static class ResultAccessTracker<R> {
 
+        /**
+         * Future that completes with the result of the asynchronous operation that this instance
+         * tracks.
+         */
+        @Nonnull private final CompletableFuture<R> operationResultFuture;
+
         /** Result of an asynchronous operation. Null if operation is in progress. */
         @Nullable private final Either<Throwable, R> operationResultOrError;
```

Comment:
This thing isn't really necessary anymore, because the future already contains everything.